### PR TITLE
Add support for Conv LoRA layers

### PR DESCRIPTION
**Pull request by RyanJDick**
Adds `LoRAConv1dLayer`, `LoRAConv2dLayer`, and `LoRAConv3dLayer` with unit tests.

- [x] Merge #7 and change the target branch to `main`

### CODE REVIEW

File: src/invoke_training/lora/layers/lora_conv_layer.py [hunk 1]
```diff
@@ -0,0 +1,131 @@
...
+        self.register_buffer("alpha", torch.tensor(alpha, device=device, dtype=dtype))
+
+        self._rank = rank
+
+        self.reset_parameters()
+
+    def reset_parameters(self):
+        # This initialization is based on Microsoft's implementation:
+        # https://github.com/microsoft/LoRA/blob/998cfe4d351f4d6b4a47f0921dec2397aa0b9dfe/loralib/layers.py#L279
+        torch.nn.init.kaiming_uniform_(self._down.weight, a=math.sqrt(5))
```
Comment (brandonrising) [inline src/invoke_training/lora/layers/lora_conv_layer.py:76]: What does this do?

**Review by brandonrising — APPROVED** [review]
lgtm!

File: src/invoke_training/lora/layers/lora_conv_layer.py (hunk 1 above)
Comment (RyanJDick) [reply src/invoke_training/lora/layers/lora_conv_layer.py:76]: I'm not exactly sure what you mean by *this*, but I'll try to explain.

`reset_parameters()` sets the initial LoRA weights before they are trained.

Different weight initializations have different convergence properties. I don't know the details of the `kaiming_uniform_` initialization, but generally the goal is to initialize the weights such that when you stack a bunch of layers in a network you don't exponentially grow/diminish the activations in the forward pass (or the gradients in the backwards pass).

I copied this `kaiming_uniform_` configuration from Microsoft's LoRA implementation, and it looks like they copied it from the [default torch.nn.Linear initialization](https://github.com/pytorch/pytorch/blob/2a138d7f1df70d7b61bdc59fd43d42f8f59a4262/torch/nn/modules/linear.py#L104-L107). (If you really want to go down a rabbit hole, follow that think for the explanation of why torch uses it.)

For LoRA layers, the 'up' weights are initialized to 0. This is done so that the LoRA layer can be added to a network and it will initially have no effect on the network behaviour. Then, as it is trained, it will gradually exert more influence over the base model.

Comment (brandonrising) [reply src/invoke_training/lora/layers/lora_conv_layer.py:76]: Gotcha thanks! I was mostly asking about the `kaiming_uniform_` function, and the reason why it's run with `a=math.sqrt(5)`

I went a little bit down the rabbithole and it seemed like it'd take me a while to understand it fully so I was curious if you had an answer quickly for it. Not really important as long as it works and is documented to be useful. That explanation for what the initialization for is plenty for me, thanks!
